Labs 2,3,4,5 submission

Anurni opened the pull request.

Anurni committed:
added child states in prompt and some actions

Anurni committed:
meetingpersonlisten finally works and updates the context

Anurni committed:
listen-speak states all work now and context gets updated correctly

Anurni committed:
implemented all speak and listen actions, does not work yet tho

Anurni committed:
added ASR_NOINPUT events, added timeout event, doesnt still display anything in localhost

Anurni committed:
started looking into the grammar, definitely does not work yet

Anurni committed:
developed the logic, added functions, sound gone suddenly?

Anurni committed:
sound works now and logic chart is almost completely implemented :)

Anurni committed:
Almost finished, only missing re-raising question when event "ASR_NOINPUT" takes place

Anurni committed:
added greeting state, ASR_NOINPUT loop still to fix

Anurni committed:
lab report (lab2)

Anurni committed:
final version (ASR_NOINPUT loop fixed)

Anurni committed:
Merge branch 'main' of https://github.com/Anurni/dialogue-systems-1-2024

RasmusBlanck commented:
Good!

Just a small comment about the time slots, it seems that the following format works:   
"8:15": {time: "08:15"},

Anurni committed:
added lab 3 report

Anurni committed:
audio file (part 2)

Added extra sound features (finger snap) with Audacity (tried to do it with Azure but didn't manage).
Really tried my best with this one. :)

Anurni committed:
SSML code for the poem

Worked on this quite awhile, had some issues with nesting some of the elements (for instance I tried to apply the <prosody pitch="low" rate="medium"></prosody> to the whole text, but for some reason it didn't work and I had to add it to each utterance separately). For some reason I also didn't hear any difference in the volume when adjusting it (for example I wanted to adjust the volume of the word "dirt" with <prosody volume="+20dB"></prosody>). Also some issues with emphasizing words, would be great if we could go through these elements once again? :)

Anurni committed:
dm4 code created

Anurni committed:
Merge branch 'main' of https://github.com/Anurni/dialogue-systems-1-2024

Anurni committed:
added pre-defined actions etc

Anurni committed:
corrected typo

Anurni committed:
started to implement the logic

Anurni committed:
some modifications to the logic, cannot log topEntity yet

Anurni committed:
states done, missing celebrity info

Anurni committed:
logic implemented, add the rest of ASR_NOINPUT states

Anurni committed:
final version

Anurni committed:
uploaded NLU-json file (lab4)

Anurni committed:
added help intent and entities

Anurni committed:
add the rest of the confidence threshold stuff, fix the help error in the beginning

Anurni committed:
implemented 3-time max repetition (but make sure it goes to 0 after each state transition!!!)

Anurni committed:
help and repetition reset to 0 DONE, fix: low confidence score guard in "Listen" state

Anurni committed:
celebrity confidence verification works, idk why meeting one doesnt, otherwise done

Anurni committed:
final version

Anurni commented:
Lab5 code works otherwise but for some reason when this guard evaluates to true:
`` {guard: ({event}) => checkIfMeetingIntent(event.nluValue.topIntent) === true && checkThreshold(event.nluValue.intents[0].confidenceScore) === false, actions: "setRepetitionBackToZero",
            target: "VerifyTheTopIntentIsMeetingSpeak"},``
I get an error that says `Uncaught (in promise) TypeError: Cannot read properties of null (reading 'reject')
    at MicAudioSource.ts:142:57`
This is weird because the equivalent guard for the WhoIsX intent works just fine!

Also, I didn't quite understand this question: "Can you implement a similar threshold for natural language understanding? How can it be combined with the ASR threshold? Justify your choice (as a comment in your code) and provide a sketch implementation." since I thought we are specifically supposed to inspect the confidence levels of the NLU model's intents.

Anurni commented:
lab 3 resource key: c6cfcc58f07142859860c92e1945fb8f

Anurni commented:
const azureLanguageCredentials = {
  endpoint: "https://annis-lab4.cognitiveservices.azure.com/language/:analyze-conversations?api-version=2022-10-01-preview",
  key: "5e24c0929d2944548c22a99ab629000c",
  deploymentName: "appointment",
  projectName: "appointment",
};

RasmusBlanck commented:
Lab 4 comments:

Well done!

The NLU for booking a meeting seems to be working very well. The interaction was generally smooth. 

The machine seems to crash if asked for a celebrity for which there is no entity on the NLU side. I worked around this by looking at the json, but it would be good to have a failsafe in place.

Suggestions for improvement: Make it possible to say, for example, "Book a meeting with Vlad" directly, and get to the "at what day and time"-question. Your NLU training data suggests that this should be possible. Also, if the user only says a day in response to the "what day and time"-question, it would be good to get another question about the desired time. You can never trust your user. ;-)

RasmusBlanck commented:
Lab 5 comments:

Mostly good work, but there are two minor issues that I'd like you to fix: 

In case you try to assign an out-of-grammar name to a meeting, the machine replies "I couldn't properly register the meeting title, sorry." and then just stops. I'm not sure exactly why it's doing that, and it feels like uninteded behaviour. (It's not really transparent exactly what kinds of meeting titles are available. In a "real" system, it would perhaps be more flexible if the meeting titles weren't constrained by a grammar. You don't have to change this, it's more of a general comment.)

If I say "Can I meet someone?" the machine just stops. There are two states named "VerifyTheTopIntentIsMeetingSpeak", which is probably the cause of this problem.

It should probably be pretty easy to fix this to get a pass.

Anurni committed:
verifythetopintentismeeting bug fixed

Anurni commented:
> 
> If I say "Can I meet someone?" the machine just stops. There are two states named "VerifyTheTopIntentIsMeetingSpeak", which is probably the cause of this problem.
> 
I fixed this issue now.
